### ISSUE TITLE
Bug: Items are not mailed when inventory is full

### ISSUE THREAD

**Issue by imevul** [azerothcore/mod-aoe-loot#31]
### Current Behaviour

When your inventory is full, items are simply shown in the loot window and you get a message saying your inventory is full.

### Expected Behaviour

Items should be mailed to you when your inventory is full

### Steps to reproduce the problem

1. Have full inventory
2. Loot item(s)

### Extra Notes

Since commit [d777d18](https://github.com/azerothcore/mod-aoe-loot/commit/d777d183c521201c46a68dd1ecf431cbc21317bc)

### AC rev. hash/commit

bf17d93727e1

### Operating system

Ubuntu 22.04.3 LTS (x64)

### Custom changes or Modules

- mod-account-achievements
- mod-accountbound
- mod-ah-bot
- mod-aoe-loot
- mod-eluna
- mod-guildhouse
- mod-learn-spells
- mod-reagent-bank-account
- mod-solo-lfg
- mod-solocraft
- mod-transmog

**Comment by pangolp** [comment]
That's correct, due to the code rewriting, this issue and the other one mentioned in the other issue, are 2 issues that currently occur. I think we have to make some changes. First there was probably the gold issue, and then I started working on this one, or maybe I did both at the same time, but I want to avoid breaking any other mechanics in the process.

**Comment by pangolp** [comment]
In blizzlike content, you only get runs when you get an item through the role and you don't have space. But if you are going to strip a ciratura and your bags are full, it won't send you emails. I was debating this topic with a friend, and in part, maybe he is right, because it will cause you to have practically infinite bags and the bags have no value. Since everything comes to you by mail. However, since it can be enabled or not, I may end up adding it anyway. By default, it will be disabled, and if you enable it, it will send the emails, but I don't know how long it will take to implement it.

**Comment by pangolp** [comment]
I don't think this mail thing will ever be implemented. Although, as it would be optional, maybe someone will do it at some point. For now, I'm trying to make the module as broad as possible, and to make it adaptable to the largest number of projects. But with the recent changes in the module, I don't know if this will really be possible to implement.

**Comment by TheSCREWEDSoftware** [comment]
> I don't think this mail thing will ever be implemented. Although, as it would be optional, maybe someone will do it at some point. For now, I'm trying to make the module as broad as possible, and to make it adaptable to the largest number of projects. But with the recent changes in the module, I don't know if this will really be possible to implement.

What happened exactly to the mailing system, was the script that does checking that broke with the newer updates of AC?

Also at least I suggestion, removal of option from the configuration (for aoe mail) and update at least the module page in github

**Comment by pangolp** [comment]
> What happened exactly to the mailing system, was the script that does checking that broke with the newer updates of AC?
> 
> Also at least I suggestion, removal of option from the configuration (for aoe mail) and update at least the module page in github

Basically, the previous module would go through the bodies and add the items to the player's inventory. That was causing some issues, but this morning, the change we made was to send the items to a main body, removing the items from the rest of the body. Basically, all the items go to a single body, but the player must take said items, he doesn't get them in the bag (unless he takes them). Previously, in the old logic, as we sent the items to him, what we did was, check if he didn't have space, and send the items to him by mail. But the current idea seems a bit cleaner, and more similar to the loot that was added back in the days in Pandaria. In fact, if the client didn't have the 18 item limitation, we could have the same system that was added in Pandaria. Still, we have to continue improving the module, and we continue working to finish it, and make it work at 100%, but some limitations are the client's, and we cannot make changes to it, at least not without making patches and editing it.

That's why I think that perhaps the best thing would be to work on this email thing in a different module and try to make it as fast as possible as the official one was in its day, because there are many validations to take into account, and we must not forget that we do not have personal loot in this expansion, all the loot usually belongs to the group, which usually brings some problems when it comes to distributing the loot.
